dask/typing.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Mapping, MutableMapping, Protocol, runtime_checkable
+
+try:
+    from IPython.display import DisplayObject
+except ImportError:
+    DisplayObject = object
+
+
+@runtime_checkable
+class DaskCollection(Protocol):
+    def __dask_graph__(self) -> Mapping | None:
+        pass
+
+    def __dask_keys__(self) -> list:
+        pass
+
+    def __dask_postcompute__(self) -> tuple[Callable, tuple]:
```

Comment:
It would be nice to include the signature of the `Callable`

Reply:
Perhaps we can create another Protocol for postcompute and postpersist callables, since those functions do have some requirements. Working on updating the PR with an idea for that.